flutter/lib/src/navigation/time_to_initial_display_tracker.dart

```diff
@@ -0,0 +1,132 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+// ignore: implementation_imports
+import 'package:sentry/src/sentry_tracer.dart';
+
+import '../../sentry_flutter.dart';
+import '../frame_callback_handler.dart';
+
+@internal
+class TimeToInitialDisplayTracker {
+  static final TimeToInitialDisplayTracker _instance =
+      TimeToInitialDisplayTracker._();
+
+  factory TimeToInitialDisplayTracker(
+      {FrameCallbackHandler? frameCallbackHandler}) {
+    if (frameCallbackHandler != null) {
+      _instance._frameCallbackHandler = frameCallbackHandler;
+    }
+    return _instance;
+  }
+
+  TimeToInitialDisplayTracker._();
+
+  FrameCallbackHandler _frameCallbackHandler = DefaultFrameCallbackHandler();
+  bool _isManual = false;
+  Completer<DateTime>? _trackingCompleter;
+  DateTime? _endTimestamp;
+
+  /// This endTimestamp is needed in the [TimeToFullDisplayTracker] class
+  @internal
+  DateTime? get endTimestamp => _endTimestamp;
+
+  Future<void> trackRegularRoute(
+    ISentrySpan transaction,
+    DateTime startTimestamp,
+  ) async {
+    await _trackTimeToInitialDisplay(
+      transaction: transaction,
+      startTimestamp: startTimestamp,
+      // endTimestamp is null by default, determined inside the private method
+      // origin could be set here if needed, or determined inside the private method
+    );
+  }
+
+  Future<void> trackAppStart(ISentrySpan transaction,
+      {required DateTime startTimestamp,
+      required DateTime endTimestamp}) async {
+    await _trackTimeToInitialDisplay(
+      transaction: transaction,
+      startTimestamp: startTimestamp,
+      endTimestamp: endTimestamp,
+      origin: SentryTraceOrigins.autoUiTimeToDisplay,
+    );
+
+    // Store the end timestamp for potential use by TTFD tracking
+    _endTimestamp = endTimestamp;
+  }
+
+  Future<void> _trackTimeToInitialDisplay({
+    required ISentrySpan transaction,
+    required DateTime startTimestamp,
+    DateTime? endTimestamp,
+    String? origin,
+  }) async {
+    // Determine endTimestamp if not provided
+    final _endTimestamp = endTimestamp ?? await determineEndTime();
```

Comment:
`h`: Shouldn't we add some timeout for this? Otherwise, this could last forever.